lib/ddtrace/contrib/extensions.rb

```diff
@@ -27,8 +27,9 @@ def configure(target = configuration, opts = {})
 
           # Activate integrations
           if target.respond_to?(:integrations_pending_activation)
+            silence_logs = target.respond_to?(:silence_logs?) ? target.silence_logs? : false
             target.integrations_pending_activation.each do |integration|
-              integration.patch if integration.respond_to?(:patch)
+              integration.patch(silence_logs) if integration.respond_to?(:patch)
```

Comment:
This is very tricky, as silencing all logs will also hide important information. For example, the instrumentation might not be compatible with the gem version installed (we support >= 1.0, but 0.9 is installed) (`Patchable.compatible?`), or the gem might not have been `require`d yet (`Patchable.loaded?`).

Both of these cases currently output a warning today.

Maybe we could have `Patchable#patch` return `true` is patching was successful, or a Hash if something failed:
```ruby
{
  available: self.class.available?,
  loaded: self.class.loaded?,
  compatible: self.class.compatible?,
  patchable: self.class.patchable?,
}
```
And let the caller handle the error reporting.

What do you think?

Reply:
makes sense, updated. I still think the logging is a little clunky but it's more flexible than previously.